lua/entities/gmod_time_distortion_generator/init.lua

```diff
@@ -18,89 +18,160 @@ function ENT:Initialize()
 
     local phys = self:GetPhysicsObject()
 
+    self.Radius = 1000
+    self.EntHealth = 100
+    self.EntMaxHealth = 100
+
     if (phys:IsValid()) then
         phys:Wake()
     end
+
+    if not WireLib then 
+        return 
+    else
+        local inNames = {"Activate","Radius"}
+	    local inTypes = {"NORMAL","NORMAL"}
+	    WireLib.CreateSpecialInputs( self,inNames,inTypes)
+        Wire_CreateOutputs(self,{"Active","Radius","Health"})
+
+        Wire_TriggerOutput(self,"Radius",1000)
+    end
 end
 
-function ENT:Use(ply)
-    if self.broken then return end
+function ENT:TriggerWire(tname,value)
+    if(not WireLib) then
+        return
+    else
+        Wire_TriggerOutput(self,tname,value)
+    end
+end
+
+function ENT:TriggerInput(iname, value)
+    if(iname == "Activate") then
+        if self.Broken then return end
+
+        if(value >= 1) then
+            if(self:GetEnabled() == false) then
+                self:TurnOn(true)
+            end
+        else
+            if(self:GetEnabled() == true) then
+                self:TurnOn(false)
+                Wire_TriggerOutput(self,"Active",0)
+            end
+        end
+    elseif(iname == "Radius") then
+        self.Radius = math.Clamp(value,0,2048)
```

Comment:
It would be good to have a `SetRadius` function similar to the TurnOn function to do this so it could be used by other things in future